packages/react-router/modules/matchPath.js

```diff
@@ -71,14 +71,14 @@ const matchPath = (pathname, options = {}) => {
 
   if (exact && !isExact) return null;
 
-  const matchParams = keys.reduce((memo, key, index) => {
-    memo[key.name] = values[index];
-    return memo;
+  const matchParams = keys.reduce((params, key, index) => {
+    params[key.name] = values[index];
+    return params;
   }, {});
 
   const params = absolute
     ? matchParams
-    : Object.assign({}, parent && parent.params, matchParams);
+    : { ...(parent && parent.params), ...matchParams };
```

Comment:
I think things would be more readable if there were a separate variable called `parentParams` containing the params from `parent.params` like so:
```javascript
const parentParams = parent ? parent.params : {};

const params = absolute
  ? matchParams
  : { ...parentParams, ...matchParams };
```

Reply:
That's really minor, IMO :)

Reply:
@karevn Indeed, but as long as the state of this PR is pending, changes can be added, IMO :)

Reply:
I gave it a second thought... And it can be rewritten the next way:
```javascript
const params = {...parentParams, ...matchParams}
```

as `{...null}` or `{...undefined}` evaluates to `{}` :)

Still want to introduce a new variable? ;)

Reply:
Feels like you are overthinking something quite a lot :) What would be actually helpful is to use suggestion feature so it can be used right away by a maintainer instead of copy & pasting it ;)

![image](https://user-images.githubusercontent.com/1096340/53301913-a2bbb980-3858-11e9-9b6d-3aba825faeda.png)

Reply:
@karevn You don't have to tell me how object destructuring in JavaScript works ;) I am aware that this current version works very well, but it's a mess to read it. Therefore I stay with my previous opinion and like to have an additional variable for this case.

@FredyC Do you mean me or @karevn in your comment?

Reply:
...'cause `parentParams` reads way better than `parent.params`. It's up to Luka anyway. Or you can create your own PR :)

Reply:
@MeiKatz Each of you can make your own suggestion :) It's always better than just arguing imo ;)

Reply:
@karevn I will wait some days and watch if this PR gets a step forward. Otherwise I will create my own PR for this topic. In this case your welcome to review it :)